<a href="https://colab.research.google.com/github/Pogayo/Machine-Learning/blob/master/E_learning_improve.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

The purpose of this notebook is to investigate how students feel about the e-learning program. I will analyze the data from the e-learning satisfaction survey, specifically the question on how the program should be improved?

**Methodology**
1. Will use sentiment analysis to gauge the polarity of the comments --Will then compare the polarity with the responses on how likely the students are to recommend the program to others.
2. I will also extract the specific courses that students had a problem wi

In [2]:
from google.colab import files
uploaded=files.upload()

Saving satisfaction.csv to satisfaction.csv


In [0]:
from textblob import TextBlob
import numpy as np
import pandas as pd
import io
import six    #will be used to check if a variable is a string
import matplotlib.pyplot as plt


In [0]:
raw_data=pd.read_csv(io.BytesIO(uploaded['satisfaction.csv']), header=[1])

In [7]:
raw_data.head()

Unnamed: 0  ... Would you explain in your own words what need to be improved in the training to increase your satisfaction level?
0         NaN  ...  what I may suggest in front office agents depa...                                                                
1         NaN  ...  For me including professional pactices like op...                                                                
2         NaN  ...                                                NaN                                                                
3         NaN  ...                                                NaN                                                                
4         NaN  ...                                                NaN                                                                

[5 rows x 41 columns]

In [24]:
improved=raw_data['Would you explain in your own words what need to be improved \ufeffin the training to increase your satisfaction level?']
improved=improved.drop_duplicates()
improved=improved.dropna()
improved=improved.rename("improve_col")
improved.describe()

count                                                   318
unique                                                  318
top       In general, the most needed is to increase the...
freq                                                      1
Name: improve_col, dtype: object

In [0]:
polarity=[]
subjectivity=[]


In [0]:
sum_polarity=0
min_improved_polarity=20
max_improved_polarity=-2

for comment in improved:
   blob=TextBlob(comment)
   pol=blob.polarity
   polarity.append(pol)
   sum_polarity+=pol
   if pol>max_improved_polarity:
     max_improved_polarity=pol
   if pol<min_improved_polarity:
     min_improved_polarity=pol
   subjectivity.append(blob.subjectivity)


In [0]:
avg_improved_polarity=sum_polarity/318


In [105]:
print("Average Polarity:"+str(avg_improved_polarity))
print("Maximum Polarity:"+str(max_improved_polarity))
print("Minimum Polarity:"+str(min_improved_polarity))

Average Polarity:0.1791349021058141
Maximum Polarity:1.0
Minimum Polarity:-0.30000000000000004


In [110]:
1.179/2*10
0.179/1*10


1.79

# How to improve E-learning findings 
1. Average Polarity:0.1791349021058141
2. Out of 10, the polarity is 5.89
3. Positivity of the comments out of 10 is 1.79

**Top 10 comments (Positive)**


**Bottom 10 comments(Negative)**

**Most mentioned courses**
